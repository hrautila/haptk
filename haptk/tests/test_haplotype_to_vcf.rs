mod common;

use crate::common::TEST_HAPLOTYPE;
use std::path::PathBuf;

#[test]
#[cfg(feature = "clap")]
fn haplotype_to_vcf() {
    let cmd = haptk::clap::SubCommand::HaplotypeToVcf {
        file: PathBuf::from(TEST_HAPLOTYPE),
        sample_name: String::from("TEST1"),
        log_and_verbosity: crate::common::silent_verbosity(),
        output: PathBuf::from("tests/results/TEST_haplotype.vcf"),
    };

    haptk::clap::run_cmd(cmd).unwrap();

    let res = std::fs::read_to_string("tests/results/TEST_haplotype.vcf").unwrap();
    let res = res.split("ID=chr9").nth(1).unwrap();
    insta::assert_yaml_snapshot!(res);
}
