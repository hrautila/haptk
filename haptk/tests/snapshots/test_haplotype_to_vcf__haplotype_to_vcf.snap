---
source: haptk/tests/test_haplotype_to_vcf.rs
expression: res
---
">\n#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tTEST1\nchr9\t18\tchr9_18_G_T\tG\tT\t.\tPASS\t.\tGT\t0\nchr9\t19\tchr9_19_A_C\tA\tC\t.\tPASS\t.\tGT\t0\nchr9\t20\tchr9_20_G_T\tG\tT\t.\tPASS\t.\tGT\t0\nchr9\t21\tchr9_21_A_C\tA\tC\t.\tPASS\t.\tGT\t0\nchr9\t22\tchr9_22_G_T\tG\tT\t.\tPASS\t.\tGT\t0\nchr9\t23\tchr9_23_A_C\tA\tC\t.\tPASS\t.\tGT\t0\nchr9\t24\tchr9_24_G_T\tG\tT\t.\tPASS\t.\tGT\t0\nchr9\t25\tchr9_25_A_C\tA\tC\t.\tPASS\t.\tGT\t0\nchr9\t26\tchr9_26_G_T\tG\tT\t.\tPASS\t.\tGT\t0\nchr9\t27\tchr9_27_A_C\tA\tC\t.\tPASS\t.\tGT\t0\nchr9\t28\tchr9_28_G_T\tG\tT\t.\tPASS\t.\tGT\t0\nchr9\t29\tchr9_29_A_C\tA\tC\t.\tPASS\t.\tGT\t0\nchr9\t30\tchr9_30_G_T\tG\tT\t.\tPASS\t.\tGT\t0\nchr9\t31\tchr9_31_A_C\tA\tC\t.\tPASS\t.\tGT\t0\nchr9\t32\tchr9_32_G_T\tG\tT\t.\tPASS\t.\tGT\t1\nchr9\t33\tchr9_33_A_C\tA\tC\t.\tPASS\t.\tGT\t0\nchr9\t34\tchr9_34_G_T\tG\tT\t.\tPASS\t.\tGT\t0\nchr9\t35\tchr9_35_A_C\tA\tC\t.\tPASS\t.\tGT\t0\nchr9\t36\tchr9_36_G_T\tG\tT\t.\tPASS\t.\tGT\t0\nchr9\t37\tchr9_37_A_C\tA\tC\t.\tPASS\t.\tGT\t0\nchr9\t38\tchr9_38_G_T\tG\tT\t.\tPASS\t.\tGT\t0\nchr9\t39\tchr9_39_A_C\tA\tC\t.\tPASS\t.\tGT\t0\nchr9\t40\tchr9_40_G_T\tG\tT\t.\tPASS\t.\tGT\t0\nchr9\t41\tchr9_41_A_C\tA\tC\t.\tPASS\t.\tGT\t0\nchr9\t42\tchr9_42_G_T\tG\tT\t.\tPASS\t.\tGT\t0\nchr9\t43\tchr9_43_A_C\tA\tC\t.\tPASS\t.\tGT\t0\nchr9\t44\tchr9_44_G_T\tG\tT\t.\tPASS\t.\tGT\t0\nchr9\t45\tchr9_45_A_C\tA\tC\t.\tPASS\t.\tGT\t0\nchr9\t46\tchr9_46_G_T\tG\tT\t.\tPASS\t.\tGT\t0\n"

